GrayBMP.cs

```diff
@@ -74,8 +74,27 @@ class GrayBMP {
       Dirty (x1, y1); Dirty (x2, y2);
    }
 
+   /// <summary>Draws a horizontal line between the given endpoints, with the given shade of gray</summary>
+   public void DrawHorizontalLine (int x1, int x2, int y, int gray) {
+      Begin ();
+      int stepX = x1 < x2 ? 1 : -1;

```

Comment:
Faster if you swap x1 and x2 in the beginning and always run the loop from left to right.

Reply:
Done